docs/src/appendix/packaging.md

```diff
@@ -0,0 +1,501 @@
+docs = true
+debug = true
+```
+
+The entire `[extras]` section is optional, in which all optional features are assumed to be disabled.
+If the section is present, any items that are not specified, are assumed to be disabled.
+
+The structure and full configuration of extras is not yet fully defined, but the following are some
+initial thoughts:
+
+* The `docs` field indicates whether or not to include documentation in the package. If `true`, then
+the docs are expected to be in the form of a static HTML site, placed under the `etc/docs` directory,
+with an `index.html` file as the main page of the docs. The docs will be compressed as a zip file,
+and stored within the package. If `false`, then whether or not `etc/docs` is present doesn't matter,
+it will not be included in the package binary.
+
+* The `debug` field indicates whether or not to include source-level debugging information in the package.
+If `true`, then the debug info/source maps/etc., will be expected under `etc/debug`. Currently, there is
+no format for this data, so this setting has no effect.
+
+#### Dependencies
+
+The final section of the manifest, is `[dependencies]`. This section specifies what dependencies this
+package has, and optionally, how to fetch them.
+
+The most basic way to express a dependency is as follows:
+
+```toml
+[dependencies]
+foo = "0.1.0"
+```
+
+This expresses a dependency on the `foo` package at version `0.1.0`. When expressed this way, any suitable
+match for this version string can satisfy this dependency. As such, the package must not contain direct
+references to specific MAST roots in `foo`, as the roots are not guaranteed to be stable across builds.
+For certain things, this flexibility may be convenient. However, if you want to be more precise, you
+can specify the content digest to match:
+
+```toml
+[dependencies]
+foo = { version = "0.1.0", digest = "..." }
+```
+
+This will look for a package that matches the given version string, and hashes to the given digest.
+
+It is also supported to create a package which vendors its dependencies. Such packages cannot be published
+to a registry, but are useful all-in-one artifacts that can be run in a VM that has no way to access
+external registries. Vendored packages are placed under `etc/vendor`, e.g. `etc/vendor/foo@0.1.0.mpkg`.
+You indicate that a dependency is vendored like so:
+
+```toml
+[dependencies]
+foo = { version = "0.1.0", vendored = true }
+```
+
+You can also specify the `digest` key, and the hash of the vendored package will be checked against the
+digest when loading the dependency.
+
+Currently, there are no other ways to specify where to fetch a package from, e.g. `git` or `url` dependencies,
+but those may come once we have more package infrastructure in place.
+
+> [!IMPORTANT] If there are dependencies specified in the WIT type definition file, then they
+> must agree with those specified in the `[dependencies]` section. This will be checked when
+> a package is created.
```

Comment:
There are `namespace` part in WIT (as in `namespace:package@version`) which is not addressed here. We could employ namespaces as well.

Reply:
I was imagining the package `name` field would be the fully-qualified name, e.g. `namespace:package`, but I think you're on to something here, and that it would make sense to have all three as separate fields, i.e. `namespace`, `name`, and `version`.

I'm not yet sure how we want to make use of namespaces, whether we want to reserve some of them, etc., but I think it makes sense to formalize it as part of the fully-qualified package name, and reserve `core`, `std`, `miden`, and `mast` for now. Thoughts?

Reply:
> I was imagining the package `name` field would be the fully-qualified name, e.g. `namespace:package`, but I think you're on to something here, and that it would make sense to have all three as separate fields, i.e. `namespace`, `name`, and `version`.

Actually, I think `namespace:package` format for our `name` field should be enough. This approach is used in `cargo-component` for WIT dependencies. 

> I'm not yet sure how we want to make use of namespaces, whether we want to reserve some of them, etc., but I think it makes sense to formalize it as part of the fully-qualified package name, and reserve `core`, `std`, `miden`, and `mast` for now. Thoughts?

I agree.